realm/src/main/java/io/realm/Realm.java

```diff
@@ -595,23 +600,66 @@ private static synchronized Realm createAndValidate(RealmConfiguration config, b
         return realm;
     }
 
+    // Make sure that the new configuration doesn't clash with any existing configurations for the Realm
+    private static void validateAgainstExistingConfigurations(RealmConfiguration newConfiguration) {
```

Comment:
This one is only called from `createAndValidate()`? And since `createAndValidate()` is `synchronized`, the operations in here are thread-safe, right?